File: fleet-pay-source-code 1 (1)/fleet-pay-export/flutter/lib/models/company_user.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'company.dart';
import 'user.dart';
import 'role.dart';
import 'user_status.dart';
// --- start import code ---
  
// --- end import code ---

part 'company_user.g.dart';
part 'company_user.freezed.dart';

@unfreezed
class CompanyUser with _$CompanyUser {
  // --- start init code ---
  
  // --- end init code ---

  factory CompanyUser({
    required String id,
    required String companyId,
    Company? company,
    required String userId,
    User? user,
    required Role role,
    required UserStatus status,
  }) = _CompanyUser;

  factory CompanyUser.fromJson(Map<String, dynamic> json) => _$CompanyUserFromJson(json);

  // --- start extension code ---
  
  // --- end extension code ---

}

File: fleet-pay-export/app/src/models/company_user.ts
import { Company } from './company';
import { User } from './user';
import { Role } from './role';
import { UserStatus } from './user_status';

export interface CompanyUser {
  id: string,
  companyId: string,
  company?: Company | null,
  userId: string,
  user?: User | null,
  role: Role,
  status: UserStatus,
  // --- start extra_fields code ---
  
  // --- end extra_fields code ---
}
